### PR TITLE
Feature events and data binding

### PR DESCRIPTION
**Pull request by RandomOutput**
- Add eventing 
- Add data binding
- Add data-binding test scene.

### CODE REVIEW

File: Assets/LeapMotion/Widgets/Scripts/Button/ButtonBase.cs [hunk 1]
```diff
@@ -1,36 +1,57 @@
...
+    // Binary Interaction Handler - Fires when interaction with the widget ends.
+    public event EventHandler<LMWidgets.EventArg<bool>> EndHandler;
+
     public float triggerDistance = 0.025f;
     public float cushionThickness = 0.005f;
-    
-    public virtual event EventHandler<WidgetEventArg<bool>> StartHandler;
-    public virtual event EventHandler<WidgetEventArg<bool>> EndHandler;
+
+    protected float scaled_spring_;
```
Comment (protodeep) [inline Assets/LeapMotion/Widgets/Scripts/Button/ButtonBase.cs]: go look at if these are really needed.

File: Assets/LeapMotion/Widgets/Scripts/Interfaces/DataBinder.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    // Grab the inital value of GetCurrentData
+    void Start() {
+      m_lastDataValue = GetCurrentData();
+    }
+
+    // Checks for change in data.
+    void Update() {
+      T currentData = GetCurrentData();
+
+      if ( compare (m_lastDataValue, currentData ) ) {
```
Comment (protodeep) [inline Assets/LeapMotion/Widgets/Scripts/Interfaces/DataBinder.cs]: Go fix this to make this !compare (...

File: Assets/LeapMotion/Widgets/Scripts/Physics/LeapPhysicsBase.cs [hunk 3]
```diff
@@ -14,11 +15,24 @@ public enum LeapPhysicsState
   /// </summary>
   public abstract class LeapPhysicsBase : MonoBehaviour
   {
-    protected LeapPhysicsState m_state = LeapPhysicsState.Reflecting;
```
Comment (protodeep) [inline Assets/LeapMotion/Widgets/Scripts/Physics/LeapPhysicsBase.cs]: Add a comment here not to set this directly.

File: Assets/LeapMotion/Widgets/Scripts/Scroll/ScrollTextBase.cs [hunk 4]
```diff
@@ -19,8 +27,37 @@ public abstract class ScrollTextBase : LeapPhysicsSpring
...
+    protected virtual void scrollPressed() {
+      fireScrollStart(content.transform.localPosition.y);
+    }
+
+    protected virtual void scrollReleased() {
+      fireScrollEnd(content.transform.localPosition.y);
+    }
+
+    protected virtual void fireScrollStart(float value) {
+      Debug.Log("Scroll Start");
```
Comment (protodeep) [inline Assets/LeapMotion/Widgets/Scripts/Scroll/ScrollTextBase.cs]: Get rid of these debug statements.

File: Assets/LeapMotion/Widgets/Scripts/Button/ButtonBase.cs (hunk 1 above)
Comment (RandomOutput) [inline Assets/LeapMotion/Widgets/Scripts/Button/ButtonBase.cs]: Done. There are references.

File: Assets/LeapMotion/Widgets/Scripts/Interfaces/DataBinder.cs (hunk 2 above)
Comment (RandomOutput) [inline Assets/LeapMotion/Widgets/Scripts/Interfaces/DataBinder.cs]: fixed. Nice improvement.

File: Assets/LeapMotion/Widgets/Scripts/Physics/LeapPhysicsBase.cs (hunk 3 above)
Comment (RandomOutput) [inline Assets/LeapMotion/Widgets/Scripts/Physics/LeapPhysicsBase.cs]: Done

**Pull request merged**